### ISSUE TITLE
Wrong inference for equals operator bug

### ISSUE THREAD

**Issue by piluc** [LorenzoBettini/javamm#39]
Consider the following code:

int a = 2;
int b = 2;
int c = 2;
if (a==b==c==2) {
	System.out.println("TRUE");
}
The control expression of the if statement should generate an error. But it doesn't. And the generated code does not compile.

**Comment by LorenzoBettini** [comment]
Something similar happens in Xtend itself

```
class MyXtendClass {
	def static m(int i, int j, int k) {
		if (i===j===k) {
			println(true)
		}
	}
}
```

What Xtend does is retrieving possible error markers in the generated Java code and report them in the original Xtend code (e.g., "Java problem: Incompatible operand types Boolean and Integer") using org.eclipse.xtext.builder.smap.DerivedResourceMarkerCopier enabled in XtendConfigurableIssueCodes:

```
	@Override
	protected void initialize(IAcceptor<PreferenceKey> iAcceptor) {
		super.initialize(iAcceptor);
...
		// overwrite xbase default
		iAcceptor.accept(create(org.eclipse.xtext.xbase.validation.IssueCodes.COPY_JAVA_PROBLEMS, SeverityConverter.SEVERITY_ERROR));
	}
```

**Comment by LorenzoBettini** [comment]
available now in the new release 1.3.2
